### PR TITLE
Discovery OKR

### PR DESCRIPTION
**Pull request by tdykstra**
Contributes to https://github.com/dotnet/docs/issues/28383

* [Classes, structs, and records](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/object-oriented/)

  Organic search 15% PV 7K. This article doesn't have a clear intended purpose. Its title is "classes, structs, and records", but its h1 is "object-oriented programming" and it's the overview node in the toc "object-oriented programming" section.  The next article in the TOC section is about "objects" as class/struct/record instances. Only 2 of 14 h2s are relevant to the object-oriented title.  Just one h2 talks about records. Most of the remaining h2s are about types or topics related to types (like the Members h2), mentioning classes and structs in the text. As a whole, the article seems most like an overview of classes, structs, and records, so that's what I'm making the title, h1, and intro paragraph consistently represent. Longer term, an overhaul of this TOC section will be worthwhile.

* [Numbers in C# - Introduction to C# interactive tutorial](https://docs.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/tutorials/numbers-in-csharp) Organic search .4% PV 6K.
* [Branches and loops - Introduction to C# interactive tutorial](https://docs.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/tutorials/branches-and-loops) Organic search .7% PV 5K.
* [Data collections - Introduction to C# interactive tutorial](https://docs.microsoft.com/en-us/dotnet/csharp/tour-of-csharp/tutorials/list-collection) Organic search 1.4% PV 4K.

  The interactive tutorials get almost no organic search traffic because all the search engine bots see is the first page, which only has an intro paragraph. The best way to increase organic search traffic for these tutorials is to focus on the corresponding local-environment tutorials and make sure that those feature prominent links to their interactive counterparts. However, any traffic increase will show up in the interactive tutorial's stats as Microsoft traffic, not organic search.

* [Identifier names](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names)

  Organic search 19% PV 4K. Added "C#" to title, h1, and toc node that links to the article. This article already shows up in the first page of results for searches such as c# names and variants like c# variable names, but not for c# naming and variants. Made "naming" more prominent by putting it in a new h2.

### CODE REVIEW

File: docs/csharp/fundamentals/coding-style/identifier-names.md [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+title: "C# identifier names"
 description: "Learn the rules for valid identifier names in the C# programming language."
-ms.date: 08/21/2018
+ms.date: 03/23/2022
 ---
-# Identifier names
+# C# identifier naming rules and conventions
 
-An **identifier** is the name you assign to a type (class, interface, struct, delegate, or enum), member, variable, or namespace. Valid identifiers must follow these rules:
+An **identifier** is the name you assign to a type (class, interface, struct, delegate, or enum), member, variable, or namespace.
```
Comment (gewarren) [inline docs/csharp/fundamentals/coding-style/identifier-names.md:8]:
```suggestion
An **identifier** is the name you assign to a type (class, interface, struct, record, delegate, or enum), member, variable, or namespace.
```

File: docs/csharp/tour-of-csharp/tutorials/arrays-and-collections.md [hunk 2]
```diff
@@ -1,15 +1,17 @@
 ---
-title: Work with collections - Introduction to C# tutorial
-description: Learn C# by exploring the List collection in this tutorial.
-ms.date: 02/05/2021
+title: Work with List\<T> - Introduction to C# tutorial
+description: Learn C# by exploring the generic List collection in this tutorial.
```
Comment (gewarren) [inline docs/csharp/tour-of-csharp/tutorials/arrays-and-collections.md:3]:
```suggestion
description: Learn C# by exploring the generic List collection type in this tutorial.
```

**Review by gewarren — COMMENTED** [review]
Left a couple comments. I'll let Bill give final approval, but the rest looks good to me.

**Review by BillWagner — APPROVED** [review]
Thanks @tdykstra 

This LGTM.

**Pull request merged**